pkg/controllermanager/controller/add.go

```diff
@@ -33,3 +36,25 @@ func AddControllersToManager(mgr manager.Manager, cfg *config.ControllerManagerC
 
 	return nil
 }
+
+// AddAllFieldIndexes adds all field indexes used by gardener-controller-manager to the given FieldIndexer (i.e. cache).
+// field indexes have to be added before the cache is started (i.e. before the manager is started)
+func AddAllFieldIndexes(ctx context.Context, i client.FieldIndexer) error {
```

Comment:
Should a test be added for this as well, since it is now exported?

Reply:
I don't think, there is anything sensible this test could verify